lms/resources/_js_config/__init__.py

```diff
@@ -418,6 +419,35 @@ def _config(self):
 
         return config
 
+    def _get_product_info(self):
+        product = self._request.product
+
+        product_info = {
+            "family": product.family,
+            "features": {
+                "groups": self._request.product.settings.groups_enabled,
+            },
+            # List of API endpoints we proxy for this product
+            "api": {},
+        }
+
+        # This should be abstracted as "lms_course_id" or similar in either
+        # LTIParams or as part of product.
+        api_course_id = self._request.lti_params.get(
```

Comment:
This is what I want to tackle in a second refactor.

We have this wart here but D2L will also introduce "custom_d2l_user_id".

For D2L I'd like that to be named "lms_user_id" in the configuration and unify a way of accessing:

```
lms_user_id 
lms_course_id
...
```

that's LMS agnostic.

Reply:
Yeah, the params are a bit tricky as you need the params to get the product, but you also need the product to get the params. I wonder if we need a two phase thing where we:

 * Get basic params
 * Get the product
 * Add more product params?
 
 Unfortunately we use the presence of things like the canvas custom fields to tip us off it's canvas. I'm sure theres a way around it.